core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -1108,6 +1137,18 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
    */
   final def sandbox: ZIO[R, Cause[E], A] = foldCauseM(ZIO.fail, ZIO.succeed)
 
+  /**
+   * Extracts the optional value, or fails with the given error 'f'.
+   */
+  def someOrFail[B, E1 >: E](e: E1)(implicit ev: A <:< Option[B]): ZIO[R, E1, B] =
+    self.flatMap(ev(_).fold[ZIO[R, E1, B]](ZIO.fail(e))(ZIO.succeed))
+
+  /**
+   * Extracts the optional value, or fails with a [[scala.NoSuchElementException]]
+   */
+  def someOrFailException[B, E1 >: NoSuchElementException](implicit ev: A <:< Option[B], ev2: E <:< E1): ZIO[R, E1, B] =
```

Comment:
This method does almost the same as `ZIO#get`, do we really need both?

Reply:
@evis, I agree with you.
I think the difference is that: someOrFailException gives you an exception in case of a failure while .get just return the unit value.

what do you think about expressing `.get` using `someOrFailException`? does that make sense? 

Something like
```scala
final def get[E1 >: E, B](implicit ev: A <:< Option[B]): ZIO[R, Unit, B] =
    self.someOrFail(()).mapError(_ => ())
```

Reply:
Good idea @dorsev